src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -177,78 +185,64 @@ class TaskExecutor {
   /// receiving
   /// thread that will each continuously try to send/receive.
 
-  // Note: This method gets run in a separate thread, so all operations
-  // on member variables must be thread safe.
-  void send_messages() {
-    int max_batch_size = 5;
-    while (active_) {
-      //      std::vector<std::pair<
-      //          int,
-      //          std::vector<std::pair<VersionedModelId,
-      //          std::shared_ptr<Input>>>>>
-      //          new_messages;
-      auto current_active_models = active_containers_->get_known_models();
-      for (auto model : current_active_models) {
-        auto containers =
-            active_containers_->get_model_replicas_snapshot(model);
-        for (auto c : containers) {
-          auto batch = c->dequeue_predictions(max_batch_size);
-          if (batch.size() > 0) {
-            // move the lock up here, so that nothing can pull from the
-            // inflight_messages_
-            // map between the time a message is sent and when it gets inserted
-            // into the map
-            std::unique_lock<std::mutex> l(inflight_messages_mutex_);
-            // std::vector<const std::vector<uint8_t>> serialized_inputs;
+  void on_new_container_added(VersionedModelId model_id, int replica_id) {
+    // TODO: Create a new model queue if this is the first connected container
+    // hosting the specified model (i.e. replica_id == 0)
+    UNUSED(model_id);
+    UNUSED(replica_id);
```

Comment:
Instead of using the UNUSED macro, comment out the argument names. E.g.

```cpp
void on_new_container_added(VersionedModelId /*model_id*/, int /*replica_id*/) {
```

This is just a style thing, I think it's a little cleaner.